### PR TITLE
adjust extent of plots and add tests

### PR DESCRIPTION
**Pull request by awirb**
* extent for imshow plots - if provided in meters requires to take 0.5*cellsize into account as rasters are provided with llcellcenter as origin
* adjust constrainToData that if data reaches to raster margin - rowsMax and colsMax index is data.shape[0]-1 or data.shape[1]-1 as rowsMin:rowsMax+1, colsMin:colsMax+1 is used as index so that when rowsMax< data.shape[0] also the row of rowsMax index is added 
* add pytest
* add check if extent is in line with nrows, ncols of data array

### CODE REVIEW

File: avaframe/out3Plot/plotUtils.py [hunk 1]
```diff
@@ -859,22 +883,193 @@ def fetchContourCoords(xGrid, yGrid, data, level):
...
+    data: numpy nd array
+        data array for which the extent should be created
+    header: dict
+        header of data array: llcenter coordinates, cellsize
+
+    Returns
+    ---------
+    extentCellCenters: list
+        extent corresponding to the cell center coordinates in meters
+    extenctCellCorners: list
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:990]: typo extenct....

File: avaframe/out3Plot/plotUtils.py [hunk 2]
```diff
@@ -859,22 +883,193 @@ def fetchContourCoords(xGrid, yGrid, data, level):
...
+        extent corresponding to the cell corner coordinates (lowerleft corner, uperright corner)
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:991]: lower left and upper right

File: avaframe/out3Plot/plotUtils.py [hunk 3]
```diff
@@ -859,22 +883,193 @@ def fetchContourCoords(xGrid, yGrid, data, level):
...
+        rowsMaxPlot + 0.5 * cellSize,
+    ]
+
+    # create extent for cell centers lower left to upper right in meters
+    extentCellCenters = [colsMinPlot, colsMaxPlot, rowsMinPlot, rowsMaxPlot]
+
+    return extentCellCenters, extentCellCorners, rowsMinPlot, rowsMaxPlot, colsMinPlot, colsMaxPlot
+
+
+def createExtentMinMax(data, header, originLLCenter=True):
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:1025]: originLLCenter is missing in docstring

File: avaframe/out3Plot/statsPlots.py [hunk 4]
```diff
@@ -332,20 +328,22 @@ def plotProbMap(avaDir, inDir, cfgFull, demPlot=False):
             # also constrain DEM to data constrained
             demConstrained = demField[rowsMin:rowsMax+1, colsMin:colsMax+1]
             # add DEM hillshade with contour lines
-            ls, CS = pU.addHillShadeContours(ax1, demConstrained, cellSize, extent)
+            ls, CS = pU.addHillShadeContours(ax1, demConstrained, cellSize, extentCellCenters)
```
Comment (fso42) [inline avaframe/out3Plot/statsPlots.py:331]: returns are not used, remove them (or replace with _)

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/out1Peak/outPlotAllPeak.py [hunk 5]
```diff
@@ -164,38 +164,36 @@ def addConstrainedDataField(fileName, resType, demField, ax, cellSize, alpha=1.0
...
-    xllcenter = raster["header"]["xllcenter"]
-    yllcenter = raster["header"]["yllcenter"]
-    rowsMinPlot = rowsMin * cellSize + yllcenter
-    rowsMaxPlot = (rowsMax + 1) * cellSize + yllcenter
-    colsMinPlot = colsMin * cellSize + xllcenter
-    colsMaxPlot = (colsMax + 1) * cellSize + xllcenter
-    extent = [colsMinPlot, colsMaxPlot, rowsMinPlot, rowsMaxPlot]
+
+    # set extent in meters using cellSize and llcenter location
+    extentCellCenters, extentCellCorners, rowsMinPlot, rowsMaxPlot, colsMinPlot, colsMaxPlot = pU.createExtent(rowsMin, rowsMax, colsMin, colsMax, dataConstrained, raster['header'])
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/out1Peak/outPlotAllPeak.py:169]: rowsMinPlot, colsMinPlot are returned by the function - so I kept them

File: avaframe/out3Plot/outDistanceTimeAnalysis.py [hunk 6]
```diff
@@ -345,15 +346,16 @@ def plotMaskForMTI(cfgRangeTime, bmaskRange, bmaskAvaRadar, bmaskAvaRadarRangesl
     # add radar range gate mask
     ax1 = fig.add_subplot(131)
     ax1.set_title("Radar range gate mask")
-    im1 = ax1.imshow(bmaskRange, extent=[x0, x1, y0, y1], origin="lower", aspect=nx / ny)
+    pU.checkExtentFormat([x0-shiftCorner, x1-shiftCorner, y0-shiftCorner, y1-shiftCorner], headerInfo['cellsize'], bmaskRange, plotName="")
+    im1 = ax1.imshow(bmaskRange, extent=[x0-shiftCorner, x1-shiftCorner, y0-shiftCorner, y1-shiftCorner], origin="lower", aspect=nx / ny)
```
[the remark this answers is not in the document]
Comment (awirb) [reply avaframe/out3Plot/outDistanceTimeAnalysis.py:350]: this was a bug - they are all the same extents - I adjusted this and added comments

File: avaframe/out3Plot/plotUtils.py [hunk 7]
```diff
@@ -428,7 +428,7 @@ def saveAndOrPlot(pathDict, outFileName, fig):
 
 
 def constrainPlotsToData(inputData, cellSize, extentOption=False, constrainedData=False, buffer=""):
-    """constrain inut raster dataset to where there is data plus buffer zone
+    """constrain input raster dataset to where there is data plus buffer zone
```
Comment (fso42) [inline avaframe/out3Plot/plotUtils.py:431]: constrainedData is not in the docstring

**Review by fso42 — APPROVED** [review]

**Comment by fso42** [comment]
Standardtests ok, apart from know avaKot

**Pull request merged**